appserver/common/container-common/src/main/java/com/sun/enterprise/container/common/impl/managedbean/ManagedBeanManagerImpl.java

```diff
@@ -472,18 +454,16 @@ private boolean bundleEligible(BundleDescriptor bundle) {
 
             // Have 299 create, inject, and call PostConstruct on managed bean
 
-            JCDIService.JCDIInjectionContext jcdiContext =
+            JCDIService.JCDIInjectionContext<?> jcdiContext =
                 jcdiService.createManagedObject(managedBeanClass, bundleDescriptor);
             callerObject = (T) jcdiContext.getInstance();
 
             // Need to keep track of context in order to destroy properly
-            Map<Object, JCDIService.JCDIInjectionContext> bundleNonManagedObjs =
+            Map<Object, JCDIService.JCDIInjectionContext<?>> bundleNonManagedObjs =
                 jcdiManagedBeanInstanceMap.get(bundleDescriptor);
             bundleNonManagedObjs.put(callerObject, jcdiContext);
 
-        } else {
-
-
+        } else if (desc != null) {
```

Comment:
Hmm, 458/440 check desc == null, so it can be null. Original code would throw NPE in this branch, now it would return null callerObject. So maybe NPE would be thrown from calling method, but not from here, so yes, Sonar cannot see it now and this method is really better than before.

Reply:
Yeah, I am unsure if this is the right fix but it was the best I could see. Maybe @pdudits has an opinion on this?

Reply:
I'd rather not have one :) But reading from larger height I see this:

1. if CDI is enabled, delegate managed bean creation to CDI
2. If CDI is not enabled, we follow old Managed Bean spec. That mandates that a bean has `@ManagedBean` annotation, and therefore a desc was likely created for it

Also, interestingly the javadoc of the method says: "desc can be null if JCDI enabled bundle.".

If you want to be completely sure, add an else branch throwing `IllegalArgumentException("CDI not enabled and no managed bean of class $managedBeanClass found")`.